interfaces/builtin/media_hub.go

```diff
@@ -0,0 +1,218 @@
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package builtin
+
+import (
+	"bytes"
+
+	"github.com/snapcore/snapd/interfaces"
+	"github.com/snapcore/snapd/release"
+)
+
+var mediaHubPermanentSlotAppArmor = []byte(`
+# Description: Allow operating as the the media-hub service. Reserved because
+# this gives privileged access to the system.
+# Usage: reserved
+
+# DBus accesses
+#include <abstractions/dbus-strict>
+
+dbus (send)
+    bus=session
+    path=/org/freedesktop/DBus
+    interface=org.freedesktop.DBus
+    member="{Request,Release}Name"
+    peer=(name=org.freedesktop.DBus, label=unconfined),
+
+# Allow binding the service to the requested connection name
+dbus (bind)
+	bus=session
+	name="org.mpris.MediaPlayer2.MediaHub",
+
+# Allow binding the service to the requested connection name
+dbus (bind)
+	bus=session
+	name="core.ubuntu.media.Service",
+
+dbus (receive, send)
+	bus=session
+	path=/com/ubuntu/media/Service{,/**}
+	interface=org.freedesktop.DBus**
+	peer=(label=unconfined),
+
+# We can always connect to ourselves
+dbus (receive)
+	bus=session
+	path=/core/ubuntu/media/Service
+	peer=(label=@{profile_name}),
+
+dbus (receive)
+    bus=session
+    interface=org.freedesktop.DBus.Introspectable
+    peer=(label=unconfined),
+
+# Allow connected clients to interact with the service
+
+# Allow connected clients to interact with the player
+dbus (receive)
+    bus=session
+    interface=org.freedesktop.DBus.Properties
+    path=/core/ubuntu/media/Service
+    peer=(label=###PLUG_SECURITY_TAGS###),
+
+dbus (receive)
+    bus=session
+    interface=org.freedesktop.DBus.Introspectable
+    peer=(label=###PLUG_SECURITY_TAGS###),
+
+dbus (receive)
+    bus=session
+    interface="core.ubuntu.media.Service{,.*}"
+    path=/core/ubuntu/media/Service
+    peer=(label=###PLUG_SECURITY_TAGS###),
+
+# Allow clients to manager Player sessions
+dbus (receive)
+    bus=session
+    path=/core/ubuntu/media/Service
+    interface=core.ubuntu.media.Service
+    member="{Create,Detach,Reattach,Destroy,CreateFixed,Resume}Session"
+    peer=(label=###PLUG_SECURITY_TAGS###),
+
+# Allow clients to pause all other sessions
+dbus (receive)
+    bus=session
+    path=/core/ubuntu/media/Service
+    interface=core.ubuntu.media.Service
+    member="PauseOtherSessions"
+    peer=(label=###PLUG_SECURITY_TAGS###),
+
+# Allow clients to query/modify service properties
+dbus (receive)
+    bus=session
+    path=/core/ubuntu/media/Service{,/**}
+    peer=(label=###PLUG_SECURITY_TAGS###),
+
+dbus (send)
+    bus=session
+    path=/core/ubuntu/media/Service
+    interface=org.freedesktop.DBus.Properties
+    member=PropertiesChanged
+    peer=(label=###PLUG_SECURITY_TAGS###),
+`)
+
+var mediaHubConnectedSlotAppArmorClassic = []byte(`
+# Allow unconfined clients to interact with the player on classic
+dbus (receive)
+    bus=session
+    path=/core/ubuntu/media/Service
+    peer=(label=unconfined),
```

Comment:
This rule is a superset of this rule in mediaHubPermanentSlotAppArmor:
```
dbus (receive, send)
	bus=session
	path=/com/ubuntu/media/Service{,/**}
	interface=org.freedesktop.DBus**
	peer=(label=unconfined),
```

It looks like you are not using this variable anywhere. Please remove.